assets/js/modules/adsense/datastore/settings.test.js

```diff
@@ -0,0 +1,351 @@
+/**
+ * modules/adsense data store: settings tests.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import { STORE_NAME } from './constants';
+import {
+	ACCOUNT_STATUS_APPROVED,
+	SITE_STATUS_ADDED,
+} from '../util/status';
+import {
+	createTestRegistry,
+	subscribeUntil,
+	unsubscribeFromAll,
+} from '../../../../../tests/js/utils';
+import { getItem, setItem } from '../../../googlesitekit/api/cache';
+import { createCacheKey } from '../../../googlesitekit/api';
+
+describe( 'modules/adsense settings', () => {
+	let apiFetchSpy;
+	let registry;
+	let dispatch;
+	let select;
+
+	const validSettings = {
+		accountID: 'pub-12345678',
+		clientID: 'ca-pub-12345678',
+		useSnippet: true,
+		accountStatus: ACCOUNT_STATUS_APPROVED,
+		siteStatus: SITE_STATUS_ADDED,
+	};
+	const error = {
+		code: 'internal_error',
+		message: 'Something wrong happened.',
+		data: { status: 500 },
+	};
+
+	beforeAll( () => {
+		API.setUsingCache( false );
+	} );
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+		dispatch = registry.dispatch( STORE_NAME );
+		select = registry.select( STORE_NAME );
```

Comment:
Hmm I'm not sure I agree. Since `select` and `dispatch` are such common language and function names one should be familiar with when using datastores, I think having the variables called exactly that indicates more clearly what they are than using any custom names, even if they are maybe more semantically correct.

Reply:
My point is more about consistency, especially when the definition and usage are separated in the file. `select` and `dispatch` are common language in the codebase, but always as functions. I can see how it might be confusing for some to see this used as functions/methods in one place and variables in another. We should avoid having `select` and `dispatch` scoped to the current store (either as variables or functions) as it is is inconsistent with other usage in the codebase.

Perhaps the best solution here might be to use the full `registry` select/dispatch in each test and pull out the selectors or actions we need. This yields the most clarity in definition (where they come from and what they are) and still concise in use. I would prefer this over other options as this approach is used in the source code as well (e.g. `canSubmitChanges`).

```js
const { isAdBlockerActive } = registry.select( STORE_NAME );
const { receiveIsAdBlockerActive } = registry.dispatch( STORE_NAME );
```

I'm not suggesting we do this everywhere, as it would actually be more code to write in cases with fewer uses, but it could help clean things up in tests where the repetition gets overly verbose.